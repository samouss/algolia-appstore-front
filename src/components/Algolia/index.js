export { default as Provider } from './Provider';
export { default as withPaginateResults } from './withPaginateResults';
export { default as withQuery } from './withQuery';
export * from './utils';
