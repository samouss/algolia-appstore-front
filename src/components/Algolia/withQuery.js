import React, { Component } from 'react';
import { getDisplayName } from 'core/utils';
import { ContextTypes } from './Provider';

const withQuery = WrappedComponent => {
  class WithQuery extends Component {

    constructor(props) {
      super(props);

      this.state = {
        query: '',
      };

      this.onChange = this.onChange.bind(this);
    }

    onChange(event) {
      const query = event.currentTarget.value;

      this.setState(() => ({
        query,
      }));

      this.context.algoliaHelper
        .setQuery(query)
        .search();
    }

    render() {
      return (
        <WrappedComponent
          {...this.props}
          {...this.state}
          onChange={this.onChange}
        />
      );
    }

  }

  WithQuery.displayName = getDisplayName(
    WrappedComponent,
    'withQuery',
  );

  WithQuery.contextTypes = ContextTypes;

  return WithQuery;
};

export default withQuery;
