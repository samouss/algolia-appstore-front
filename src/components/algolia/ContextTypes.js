import PropTypes from 'prop-types';

export default {
  algoliaClient: PropTypes.object.isRequired,
  algoliaHelper: PropTypes.object.isRequired,
};
