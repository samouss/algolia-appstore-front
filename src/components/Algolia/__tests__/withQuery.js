import React from 'react';
import { shallow } from 'enzyme';
import { createMockAlgoliaClient, createMockAlgoliaHelper } from 'test/algolia';
import withQuery from '../withQuery';

describe('algolia', () => {
  describe('withQuery', () => {
    const createContext = () => ({
      algoliaHelper: createMockAlgoliaHelper(),
      algoliaClient: createMockAlgoliaClient(),
    });

    const Component = props => (
      <input {...props} />
    );

    it('expect to render', () => {
      const context = createContext();

      const props = {
        className: 'sample-class-name',
      };

      const ApplyComponent = withQuery(Component);

      const component = shallow(
        <ApplyComponent
          {...props}
        />,
        { context },
      );

      expect(component).toMatchSnapshot();
    });

    it('expect to call onChange', () => {
      const context = createContext();
      const ApplyComponent = withQuery(Component);
      const onChange = jest.spyOn(ApplyComponent.prototype, 'onChange');

      const component = shallow(
        <ApplyComponent />,
        { context },
      );

      component
        .find('Component')
        .simulate('change');

      expect(onChange).toHaveBeenCalled();
      onChange.mockRestore();
    });

    it('expect to have correct displayName', () => {
      const ApplyComponent = withQuery(Component);

      const expectation = 'withQuery(Component)';
      const actual = ApplyComponent.displayName;

      expect(actual).toBe(expectation);
    });

    describe('onChange', () => {
      it('expect to call setState', () => {
        const context = createContext();
        const ApplyComponent = withQuery(Component);

        const expectation = {
          query: 'This is a query',
        };

        const component = shallow(
          <ApplyComponent />,
          { context },
        );

        component.instance().onChange('This is a query');

        expect(component.state()).toEqual(expectation);
      });

      it('expect to call setQuery and search', () => {
        const context = createContext();
        const ApplyComponent = withQuery(Component);

        const expectation = 'This is a query';

        const component = shallow(
          <ApplyComponent />,
          { context },
        );

        component.instance().onChange('This is a query');

        expect(context.algoliaHelper.setQuery).toHaveBeenCalledWith(expectation);
        expect(context.algoliaHelper.search).toHaveBeenCalled();
      });
    });
  });
});
