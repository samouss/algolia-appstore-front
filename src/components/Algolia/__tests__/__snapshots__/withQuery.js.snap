// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`algolia withQuery expect to render 1`] = `
<Component
  className="sample-class-name"
  onChange={[Function]}
  query=""
/>
`;
